### ISSUE TITLE
Remove build artifacts after the build process?

### ISSUE THREAD

**Issue by tomaka** [emk/heroku-buildpack-rust#17]
Heroku [has a limit of 300 MB](https://devcenter.heroku.com/articles/slug-compiler#slug-size) for the slug size (ie. the output of the build process).

Unfortunately I've run into an annoying problem this morning, which is that running `cargo build --release` creates a `target` directory that weights 310 MB, though the actual binary itself is only 41 MB.

Heroku's docs suggest that you use a `.slugignore` file (similar to a `.gitignore`). However the files that match .slugignore are removed *before* the build process, so it's useless in my situation.

I don't know if you have a solution to suggest for me and people who run in the same situation as me? Removing the `/target/release/deps` in the buildpack would be nice, but as I'm not familiar with heroku's caching process I don't know if it's a good idea to do so.

**Comment by emk** [comment]
I'll need to look at this. Thanks for letting me know!

**Comment by steveklabnik** [comment]
This is hitting me today too. I think it's because this isn't properly using the cache dirs. My slugs grow and grow until they hit 300 and then fail. i have some preliminary findings here https://github.com/rust-lang-nursery/thanks/issues/31

**Comment by emk** [comment]
Entirely possible! A lot of this code was written for much older versions
of the toolchain and it still doesn't use all the options available with
the newer toolchain perfectly.

Le 25 janv. 2017 7:00 PM, "Steve Klabnik" <notifications@github.com> a
écrit :

**Comment by emk** [comment]
Hmm. On further reflection, I think the issue here may be that we don't
know the name of the built binary and so we're copying too much of target/
into the slug. I'm pretty sick this week, so I probably can't look right
away.

Le 25 janv. 2017 8:33 PM, "Eric Kidd" <eric@randomhacks.net> a écrit :

Entirely possible! A lot of this code was written for much older versions
of the toolchain and it still doesn't use all the options available with
the newer toolchain perfectly.

Le 25 janv. 2017 7:00 PM, "Steve Klabnik" <notifications@github.com> a
écrit :

**Comment by steveklabnik** [comment]
Some more build output: https://travis-ci.org/rust-lang-nursery/thanks/builds/195415786#L436-L450

>  I'm pretty sick this week, so I probably can't look right away.

❤️ don't worry about it; it's easy to deal with in the meantime, and I might look too. Get better!

**Comment by tomaka** [comment]
> it's easy to deal with in the meantime

I wonder what you mean by that.
I stopped using heroku because of this issue.

**Comment by steveklabnik** [comment]
For me;

```
$ heroku plugins:install heroku-repo
$ heroku repo:purge_cache
$ heroku repo:reset
$ git push heroku master
```

this basically re-builds your slug from scratch. since my issue was that
each build was well under 300mb, but the sum total was, this resets that
and gives me more builds.

**Comment by emk** [comment]
I think this is fixed by #22, which was just merged. Could somebody please try again and let me know if they encounter any problems?

**Comment by emk** [comment]
OK, I believe that this issue has been fixed for quite a while now, thanks to the improved target-copying code. Please reopen this if you're still seeing any problems.
